src/connector/benches/parser.rs

```diff
@@ -0,0 +1,131 @@
+                        convert_case("bravo") => json!(2),
+                        convert_case("charlie") => json!(3),
+                        convert_case("delta") => json!(4),
+                    };
+                    serde_json::to_string(&value).unwrap().as_bytes().to_vec()
+                }
+                _ => unreachable!(),
+            });
+        }
+        input.push(input_inner);
+    }
+    input
+}
+
+fn create_parser(
+    chunk_size: usize,
+    chunk_num: usize,
+    mode: &str,
+) -> (JsonParser, Vec<SourceColumnDesc>, Vec<Vec<Vec<u8>>>) {
+    let desc = vec![
+        SourceColumnDesc::simple("alpha", DataType::Int16, 0.into()),
+        SourceColumnDesc::simple("bravo", DataType::Int32, 1.into()),
+        SourceColumnDesc::simple("charlie", DataType::Int64, 2.into()),
+        SourceColumnDesc::simple("delta", DataType::Int256, 3.into()),
+    ];
+    let parser = JsonParser::new(desc.clone(), Default::default()).unwrap();
+    let input = gen_input(mode, chunk_size, chunk_num);
+    (parser, desc, input)
+}
+
+async fn parse(parser: JsonParser, column_desc: Vec<SourceColumnDesc>, input: Vec<Vec<Vec<u8>>>) {
+    for input_inner in input {
+        let mut builder =
+            SourceStreamChunkBuilder::with_capacity(column_desc.clone(), input_inner.len());
+        for payload in input_inner {
+            let row_writer = builder.row_writer();
+            parser.parse_inner(payload, row_writer).await.unwrap();
+        }
+        builder.finish();
+    }
+}
+
+fn bench_parse_match_case(c: &mut Criterion) {
+    const TOTAL_SIZE: usize = 1024 * 1024usize;
+    let rt = Runtime::new().unwrap();
+
+    for chunk_size in &[32, 128, 512, 1024, 2048, 4096] {
+        c.bench_with_input(
+            BenchmarkId::new("ParseMatchCase", chunk_size),
+            chunk_size,
+            |b, &chunk_size| {
+                let chunk_num = TOTAL_SIZE / chunk_size;
+                b.to_async(&rt).iter_batched(
+                    || create_parser(chunk_size, chunk_num, "match"),
+                    |(parser, column_desc, input)| parse(parser, column_desc, input),
+                    BatchSize::SmallInput,
+                );
+            },
+        );
+    }
+}
+
+fn bench_parse_not_match_case(c: &mut Criterion) {
+    const TOTAL_SIZE: usize = 1024 * 1024usize;
+    let rt = Runtime::new().unwrap();
+
+    for chunk_size in &[32, 128, 512, 1024, 2048, 4096] {
+        c.bench_with_input(
+            BenchmarkId::new("ParseMisMatchCase", chunk_size),
+            chunk_size,
+            |b, &chunk_size| {
+                let chunk_num = TOTAL_SIZE / chunk_size;
+                b.to_async(&rt).iter_batched(
+                    || create_parser(chunk_size, chunk_num, "mismatch"),
+                    |(parser, column_desc, input)| parse(parser, column_desc, input),
+                    BatchSize::SmallInput,
+                );
+            },
+        );
+    }
+}
```

Comment:
Seems like the code is the same for `bench_parse_match_case` and `not_match_case`? Can just create another function, parameterized by matching type.